pkg/recipes/terraform/config/config.go

```diff
@@ -40,39 +40,10 @@ const (
 
 // New creates TerraformConfig with the given module name and its inputs (module source, version, parameters)
 // Parameters are populated from environment recipe and resource recipe metadata.
-func New(ctx context.Context, moduleName string, envRecipe *recipes.EnvironmentDefinition, resourceRecipe *recipes.ResourceMetadata, envConfig *recipes.Configuration) (*TerraformConfig, error) {
-	path := envRecipe.TemplatePath
-
-	if envConfig != nil {
-		// Retrieving the secret store with associated with the template path.
-		// appends an URL prefix to the templatePath if secret store exists.
-		secretStore, err := recipes.GetSecretStoreID(*envConfig, envRecipe.TemplatePath)
-		if err != nil {
-			return nil, err
-		}
-
-		if secretStore != "" {
-			// Retrieving the URL prefix, prefix will be in the format of https://<environment>-<application>-<resource>-
-			prefix, err := recipes.GetURLPrefix(resourceRecipe)
-			if err != nil {
-				return nil, err
-			}
-
-			url, err := recipes.GetGitURL(envRecipe.TemplatePath)
```

Comment:
I think `GetGitURL` was added in `recipe/types.go` previously to prevent circular dependency. Now with this code being removed, should we consolidate all the relevant code in a single place?

Also, should `gitconfig.go` be under terraform package or do we anticipate it being shared with other drivers?

Reply:
thanks for catching this, you are right. I moved it to gitconfig. and toay we are calling `addSecretsToGitConfig` from driver so having this file in the driver, but i dont think this will be shared with other drivers. is it good to move this logic inside execute?

Reply:
thanks. Yeah I think moving it closer to terraform specific interactions would make more sense

Reply:
Discussed offline and decided to keep it as is for now, we will revisit it as part of containerization of drivers.